charts/spire/charts/spire-server/templates/tests/test-connection.yaml

```diff
@@ -28,13 +31,43 @@ spec:
           fi
           exit 1
       securityContext:
-        {{- toYaml .Values.securityContext | nindent 8 }}
+        {{- toYaml .Values.testSecurityContext | nindent 8 }}
     {{- if eq (.Values.federation.enabled | toString) "true" }}
-    - name: curl-federation-bundle-endpoint
+    - name: federation
       image: {{ template "spire-lib.image" (dict "image" .Values.tests.bash.image "global" .Values.global) }}
-      command: ['curl']
-      args: ['-k', '-s', '-f', 'https://{{ include "spire-server.fullname" . }}.{{ include "spire-server.namespace" . }}.svc.{{ include "spire-lib.cluster-domain" . }}:{{ .Values.federation.bundleEndpoint.port }}']
+      command:
+        - bash
+      args:
+        - -c
+        - |
+          # Install tools
+          apk add --update jq curl
```

Comment:
apk'ing during runtime doesn't work so well in isolated environments. Then you need a container mirror and an apk one...

I'd recommend changing the script to maybe do the following:
use 2 containers. one with the kubectl image. Do what kubectl bits you need and store the data in an emptydir within the pod in an initContainer.
2. This image seems to have curl, bash, and jq: https://edu.chainguard.dev/chainguard/chainguard-images/reference/slim-toolkit-debug/overview/  can maybe use that for the rest of the test.

Reply:
good find on the image